database/config.md
# config

## Description

<details>
<summary><strong>Table Definition</strong></summary>

```sql
CREATE TABLE `config` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `name` varchar(255) NOT NULL,
  `value` varchar(255) NOT NULL,
  `updated_at` datetime DEFAULT NULL,
  `created_at` datetime DEFAULT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `name` (`name`)
) ENGINE=InnoDB AUTO_INCREMENT=[Redacted by tbls] DEFAULT CHARSET=latin1 COLLATE=latin1_swedish_ci
```

</details>

## Columns

| Name | Type | Default | Nullable | Extra Definition | Children | Parents | Comment |
| ---- | ---- | ------- | -------- | ---------------- | -------- | ------- | ------- |
| id | int(11) |  | false | auto_increment |  |  |  |
| name | varchar(255) |  | false |  |  |  |  |
| value | varchar(255) |  | false |  |  |  |  |
| updated_at | datetime | NULL | true |  |  |  |  |
| created_at | datetime | NULL | true |  |  |  |  |

## Constraints

| Name | Type | Definition |
| ---- | ---- | ---------- |
| name | UNIQUE | UNIQUE KEY name (name) |
| PRIMARY | PRIMARY KEY | PRIMARY KEY (id) |

## Indexes

| Name | Definition |
| ---- | ---------- |
| PRIMARY | PRIMARY KEY (id) USING BTREE |
| name | UNIQUE KEY name (name) USING BTREE |

## Relations

![er](config.svg)

---

> Generated by [tbls](https://github.com/k1LoW/tbls)


CHANGELOG.md
# version : 26/07/2020
- added lock/remove event
- added version file
- added support for 30/90 day updated clients export
- added changelog.md
- add stock has been reworked to have a better UI
- fixed update last_bill not getting updated
- fixed an import issue with IDPRAK being case and we checked case insensitive
- tooth now is more modular, and includes cat & horse …
- improved tooth fiche UI
- modified stock to have a state
- removed the link to edit prices for vets
- fixed a bug where moved products would not get the right state
- separated lost from active animals
- finished rework on stock UI
- fix death/gone pets can't have fiche/consults
- add stock detail spread
- lost pets don't need to be shown as other pets
- invoice overview now shows clients last name
- pets can now be search with a like
- events can now be locked only when no report is written
- birthday is now visible on event page (for vaccines)
- slightly reworked add pet to show what is required
- added last bill to pet name search
- added product name to title when in edit mode
- changed event report UI
- added last 1m/6m/1y usage to product edit page

# version : 07/08/2020
- migrations version : 6
- added edit price to the events/bill system
- simple reductions added
- add support for noview history
- add support for changing pets from owner
- removed edit client from pets fiche
- moved edit price button so its only visible when prod/proc are added
- added a show no-history function to history (for no-history lookups)
- add report for usage (all products)
- add report for specific stock for product
- add report for specific usage for product
- add index link for reports
- add stock maintenance queries
- add stock state, and only show them (might be incomplete)
- add support for search on client id
- add vaccination report
- fixed small link bug on owner/index

# version : 16/08/2020
- migrations version : 7
- modified ui for lost & gone to take up less space, and no space if = 0
- added support for (sliced) uploading files (office, images, videos) to events;
- small error in migrations 006
- changed sql engine to InnoDB

# version : 27/08/2020
- added caching to charts
- removed client charts

# version : 14/12/2020
- add support for GS-1 DataMatrix scanning (on many veternarien products QR like code)
- header alert on empty set
- adding stock of a non existing product
- added support for Kluwer book keeping (export format : xml)
- fix lotnr (not an int)
- add support for lookup/search per breed
- simplify code
- rough selection of invoices
- improved admin invoice screen (default : 3M, until: today)
- hidden non functional messages & alerts from header
- change search behaviour, only search right side
- fix bug where name of a vet resulted in error
- small fix for owner model : now owners will only list 'active pets' (no dead or lost)
- bugfix : added result check in stock_model
- auto add month date in invoice list
- added support for nutrition advice
- designed create/edit pet page
- wysiwyg editor on report & tooth view

application/views/admin/configuration.php
<div class="row">
	<div class="col-lg-12 mb-4">

      <div class="card shadow mb-4">
			<div class="card-header d-flex flex-row align-items-center justify-content-between">
				<div><a href="<?php echo base_url('accounting/dashboard'); ?>"><?php echo $this->lang->line('admin'); ?></a> / <?php echo $this->lang->line('settings_screen'); ?></div>
				<div class="dropdown no-arrow">
				</div>
			</div>
            <div class="card-body">
			<form method="post" action="<?php echo base_url('admin/settings'); ?>" autocomplete="integrations">
				<h4>Bills</h4>
				
				<div class="form-row">
					<div class="form-group col-md-6">
						<label for="due_date">Due date</label>
						<input type="text" class="form-control" id="due_date" name="conf_due_date" value="<?php echo (isset($config['due_date'])) ? base64_decode($config['due_date']) : ''; ?>" autocomplete="due_date">
						<small id="due_datehelp" class="form-text text-muted">Bills are due, this many days. (default : 30)</small>
					</div>
				</div>
				
				<fieldset class="form-group row">
					<legend class="col-form-label col-sm-2 float-sm-left pt-0">Invoice ID Prefix</legend>
					<div class="col-sm-10">
					<div class="form-check">
						<input class="form-check-input" type="radio" name="conf_invoice_prefix" id="gridRadios1" value="YYYY" <?php echo (isset($config['invoice_prefix']) && base64_decode($config['invoice_prefix']) == "YYYY") ? 'checked' : ''; ?>>
						<label class="form-check-label" for="gridRadios1">
						Full Year (YYYY)
						</label>
					</div>
					<div class="form-check">
						<input class="form-check-input" type="radio" name="conf_invoice_prefix" id="gridRadios2" value="none" <?php echo (isset($config['invoice_prefix']) && base64_decode($config['invoice_prefix']) == "none") ? 'checked' : ''; ?>>
						<label class="form-check-label" for="gridRadios2">
						None
						</label>
					</div>
					</div>
				</fieldset>
				<h4>Events</h4>
				
				<div class="form-row">
					<div class="form-group col-md-6">
						<label for="autoclose">Autoclose reports</label>
						<input type="text" class="form-control" id="autoclose" name="conf_autoclose" value="<?php echo (isset($config['autoclose'])) ? base64_decode($config['autoclose']) : ''; ?>" autocomplete="autoclose">
						<small id="autoclosehelp" class="form-text text-muted">Reports that are never finished get auto closed after x days. (default : 14)</small>
					</div>
				</div>

				<h4>Integrations</h4>
				<!--
				<h5>Covetrus, Nerum (NetOrder)</h5>
				<hr />
				<div class="form-row">
					<div class="form-group col-md-5">
						<label for="last_name">Covetrus login</label>
						<input type="text" class="form-control" id="last_name" name="conf_covetrus_login" value="<?php echo ($config['covetrus_login']) ? base64_decode($config['covetrus_login']) : ''; ?>" autocomplete="covetrus">
						<small id="emailHelp" class="form-text text-muted">Currently not used</small>
					</div>
					<div class="form-group col-md-5">
						<label for="first_name">Covetrus pasword</label>
						<input type="password" class="form-control" name="conf_covetrus_password" value="" autocomplete="covetruspasw">
					</div>
				</div>
				-->
				<h5>MediLab</h5>
				<hr />
				<div class="form-row">
					<div class="form-group col-md-5">
						<label for="medilab_user">Medilab login</label>
						<input type="text" class="form-control" id="medilab_user" name="conf_medilab_user" value="<?php echo (isset($config['medilab_user'])) ? base64_decode($config['medilab_user']) : ''; ?>" autocomplete="medilabuser">
						<small id="medilab_userhelp" class="form-text text-muted">Used to synchronize lab results</small>
					</div>
					<div class="form-group col-md-5">
						<label for="medilab_pasw">Medilab pasword</label>
						<input type="password" class="form-control" id="medilab_pasw" value="<?php echo (isset($config['medilab_pasw'])) ? base64_decode($config['medilab_pasw']) : ''; ?>" name="conf_medilab_pasw" autocomplete="medilabpasw">
					</div>
					<div class="form-group col-md-1">
						<label for="test_covetrus">Connection</label><br/>
						<button type="button" name="test_covetrus" value="test" id="test_covetrus" class="btn btn-outline-primary">Test</button>
					</div>
				</div>
				<button type="submit" name="submit" value="edit" class="btn btn-primary">Update</button>
			</form>
			</div>
		</div>

	</div>
      
</div>

<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function(){
	$("#admin").addClass('active');

	$("#test_covetrus").on( "click", function(e) {
		$(this).html("<i class='fas fa-sync fa-spin'></i>");
		$.ajax({
			method: 'GET',
			url: '<?php echo base_url(); ?>tests/test_connection_covetrus/' + $("#medilab_user").val() + "/" + $("#medilab_pasw").val(),
			success: function(data) {
				$("#test_covetrus").html(data);
			}
    	});
	});
});
</script>
  